### PR TITLE
Add IPMI Serial definitions

### PR DESCRIPTION
**Pull request by gtossk2**
# Description

Add definitions for IPMI Serial and it can be used in ManageabilityPkg.

## How This Was Tested

Tested in https://github.com/tianocore/edk2-platforms/pull/240

## Integration Instructions

N/A

### CODE REVIEW

**Review by abdattar — COMMENTED** [review]
Code changes looks good.

File: MdePkg/Include/IndustryStandard/IpmiSerial.h
Comment (changab) [inline MdePkg/Include/IndustryStandard/IpmiSerial.h:24]: Could you please replace BM_ with BASIC_MODE_

**Review by changab — APPROVED** [review]

File: MdePkg/Include/IndustryStandard/IpmiSerial.h
Comment (gtossk2) [reply MdePkg/Include/IndustryStandard/IpmiSerial.h]: Acked.

**Comment by changab** [comment]
@gtossk2 The quiet period has now ended, however, we still need another approval.

File: MdePkg/Include/IndustryStandard/IpmiSerial.h
Comment (mdkinney) [inline MdePkg/Include/IndustryStandard/IpmiSerial.h:1]: I see Ipmi.h #includes all the IPMI related files.  Why is this one not included in Ipmi.h?

Comment (gtossk2) [reply MdePkg/Include/IndustryStandard/IpmiSerial.h:1]: Ipmi.h includes the command-related files. IpmiSerial.h is IPMI related interface, such as kcs, ssif, or serial. So I didn't include it in Ipmi.h.

Comment (changab) [reply MdePkg/Include/IndustryStandard/IpmiSerial.h:1]: > I see Ipmi.h #includes all the IPMI related files. Why is this one not included in Ipmi.h?
Hi @mdkinney 
Ipmi.h includes those protocol layer header files. IpmiSerial.h is for the transport interface which is a separate header file similar to IpmiKcs and IpmiSsif.

**Review by abdattar — APPROVED** [review]

File: MdePkg/Include/IndustryStandard/IpmiSerial.h
Comment (changab) [reply MdePkg/Include/IndustryStandard/IpmiSerial.h:1]: @mdkinney Could you please check this again and approve it if you have no concerns about this PR?
Thanks

**Comment by lgao4** [comment]
> @gtossk2 The quiet period has now ended, however, we still need another approval.

Are those PCDs really required to be configured? I want to confirm whether there is the real  usage model to configure those PCDs. If no, PCDs are not required to be introduced.

**Comment by gtossk2** [comment]
> > @gtossk2 The quiet period has now ended, however, we still need another approval.
> 
> Are those PCDs really required to be configured? I want to confirm whether there is the real usage model to configure those PCDs. If no, PCDs are not required to be introduced.

Hi @lgao4, Those PCDs based on IPMI spec should be configured per platform requirement. They are using in this PR - https://github.com/tianocore/edk2-platforms/pull/240

**Comment by lgao4** [comment]
> > > @gtossk2 The quiet period has now ended, however, we still need another approval.
> > 
> > 
> > Are those PCDs really required to be configured? I want to confirm whether there is the real usage model to configure those PCDs. If no, PCDs are not required to be introduced.
> 
> Hi @lgao4, Those PCDs based on IPMI spec should be configured per platform requirement. They are using in this PR - [tianocore/edk2-platforms#240](https://github.com/tianocore/edk2-platforms/pull/240)

Do you see the real usage case to configure those PCDs on the platform?

File: MdePkg/MdePkg.dec
Comment (mdkinney) [inline MdePkg/MdePkg.dec:2452]: Should be PcdIpmiSerialResponderLun to follow naming conventions.

Comment (gtossk2) [reply MdePkg/MdePkg.dec]: Acked.

File: MdePkg/MdePkg.dec
Comment (gtossk2) [reply MdePkg/MdePkg.dec]: Acked.

**Pull request merged**
